lang_qc/db/helper/qc.py

```diff
@@ -422,11 +422,18 @@ def _map_to_qc_state_models(
     """
     response = dict()
     for product in seq_products:
-        response[product.id_product] = []
-        for qc in product.qc_state:
-            if sequencing_outcomes_only and (qc.qc_type.qc_type != "sequencing"):
-                continue
-            response[product.id_product].append(QcState.from_orm(qc))
+        # qc_states = []
+        # for qc in product.qc_state:
+        #    if sequencing_outcomes_only and (qc.qc_type.qc_type != "sequencing"):
+        #        continue
+        #    qc_states.append(QcState.from_orm(qc))
+        qc_states = [
+            QcState.from_orm(qc)
+            for qc in product.qc_state
+            if (not sequencing_outcomes_only) or (qc.qc_type.qc_type == "sequencing")
+        ]
+        if len(qc_states) != 0:
```

Comment:
This changes the function signature, and quite possibly any API endpoint signatures too. From a REST point of view it's better to get `[]` than `None`, because you don't need to do checks before iterating.

I suspect this also qualifies as premature optimisation. I like the readability of the original better.

Reply:
This commit just fixes an inconsistency between how `get_qc_states_by_id_product_list` is described and what it does. `get_qc_states_by_id_product_list` never returned a dictionary with keys corresponding fully to a set of given product IDs.

Reply:
Better update the type hint for the return value then.